3_train.smk

```diff
@@ -1,21 +1,38 @@
-
-configfile: "3_train/train_config.yaml"
+import yaml
 
 # Save config files to output folder
 rule save_config:
     input:
         process_config = "2_process/process_config.yaml",
-        train_config = "3_train/train_config.yaml"
+        # location of training config depends on data_source and run_id
+        train_config = "3_train/in/{data_source}/{run_id}.yaml"
     output:
-        process_config = "3_train/out/{data_source}/{run_id}/process_config.yaml",
-        train_config = "3_train/out/{data_source}/{run_id}/train_config.yaml"
+        process_config = "3_train/out/{data_source}/{run_id}/{run_id}_process.yaml",
```

Comment:
I see your point. One purpose of this rule is as a quick way to keep track of what settings were used for a given model run. Since `process_config.yaml` could change from one run to another, it might be handy to keep this prefix in place. For example, if I wanted to compare config files among runs, it would be convenient for the files to be named according to their run ID.

Reply:
Sounds good - that's a good setup if `process_config.yaml` may change from run to run in the future.